model/src/main/scala/lucuma/schemas/decoders/VisitDecoders.scala

```diff
@@ -122,21 +122,25 @@ trait VisitDecoders:
 
   given decoderExecutionVisitsGmosNorth: Decoder[ExecutionVisits.GmosNorth] = Decoder.instance: c =>
     for
-      staticConfig <- c.downField("config").downField("static").as[StaticConfig.GmosNorth]
+      staticConfig <-
+        c.downField("config").downField("gmosNorth").downField("static").as[StaticConfig.GmosNorth]
       visits       <- c.downField("visits").downField("matches").as[List[Visit.GmosNorth]]
     yield ExecutionVisits.GmosNorth(staticConfig, visits)
 
   given decoderExecutionVisitsGmosSouth: Decoder[ExecutionVisits.GmosSouth] = Decoder.instance: c =>
     for
-      staticConfig <- c.downField("config").downField("static").as[StaticConfig.GmosSouth]
+      staticConfig <-
+        c.downField("config").downField("gmosSouth").downField("static").as[StaticConfig.GmosSouth]
       visits       <- c.downField("visits").downField("matches").as[List[Visit.GmosSouth]]
     yield ExecutionVisits.GmosSouth(staticConfig, visits)
 
-  given Decoder[ExecutionVisits] = Decoder.instance: c =>
-    c.downField("config")
-      .downField("instrument")
-      .as[Instrument]
-      .flatMap:
-        case Instrument.GmosNorth => c.as[ExecutionVisits.GmosNorth]
-        case Instrument.GmosSouth => c.as[ExecutionVisits.GmosSouth]
-        case _                    => DecodingFailure("Only Gmos supported", c.history).asLeft
+  given Decoder[Option[ExecutionVisits]] = Decoder.instance: c =>
+    if (c.downField("config").downField("instrument").failed) None.asRight
```

Comment:
Not sure if there is a better approach for this, but I want a None based on whether or not the `config` contains anything.

Reply:
I don't know if it's a simpler solution but this seems to work:

```scala
  given Decoder[Option[ExecutionVisits]] = Decoder.instance: c =>
    c.getOrElse[Json]("config")(Json.obj())
      .flatMap:
        _.hcursor
          .downField("instrument")
          .as[Instrument]
          .flatMap:
            case Instrument.GmosNorth => c.as[ExecutionVisits.GmosNorth].map(_.some)
            case Instrument.GmosSouth => c.as[ExecutionVisits.GmosSouth].map(_.some)
            case _                    => DecodingFailure("Only Gmos supported", c.history).asLeft
```